#!/usr/bin/env python3
PACKAGE = "follow_line_pkg"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("thresh", int_t, 0, "Thresh Value", 230, 0, 255)
gen.add("speed", double_t, 0, "Car Speed", 1.0, 0.0, 4.0)
gen.add("enable_drive",            bool_t,   0, "Enable",       False)
gen.add("ellipse", int_t, 0, "Ellipse", 13,0,25)
gen.add("medBlur", int_t, 0, "Median Blur", 1, 0, 25)
gen.add("length", int_t, 0, "length", 200, 0, 1000)
gen.add("thresh_shadow", int_t, 0, "length", 240, 0, 1000)
gen.add("alpha", double_t, 0, "alpha", 1.0, 0, 15)
gen.add("beta", double_t, 0, "beta", 20, 0, 80)
gen.add("gamma", double_t, 0, "gamma", 4, 0, 30)
gen.add("ellipse_shadow", int_t, 0, "Ellipse", 8,0,25)
gen.add("blur_shadow", int_t, 0, "Median Blur", 5, 0, 25)
exit(gen.generate(PACKAGE, "follow_line_pkg", "FollowLane"))
# The 3rd param "StopAtCw" must be this filename